pyro/contrib/oed/eig.py

```diff
@@ -147,6 +147,69 @@ def naive_rainforth_eig(model, design, observation_labels, target_labels=None,
     return (conditional_lp - marginal_lp).sum(0)/N
 
 
+def accelerated_rainforth_eig(model, design, observation_labels, target_labels,
+                              yspace, N=100, M_prime=None):
+    """
+    Accelerated Rainforth (i.e. Unnested Monte Carlo) estimate of the expected information
+    gain (EIG). The estimate is
+
+    .. math::
+TO DO ################################################
+        \\frac{1}{N}\\sum_{n=1}^N \\log p(y_n | \\theta_n, d) -
+        \\log \\left(\\frac{1}{M}\\sum_{m=1}^M p(y_n | \\theta_m, d)\\right)
+
+    Monte Carlo estimation is attempted for the :math:`\\log p(y | \\theta, d)` term if
+    the parameter `M_prime` is passed. Otherwise, it is assumed that that :math:`\\log p(y | \\theta, d)`
+    can safely be read from the model itself.
+
+    :param function model: A pyro model accepting `design` as only argument.
+    :param torch.Tensor design: Tensor representation of design
+    :param list observation_labels: A subset of the sample sites
+        present in `model`. These sites are regarded as future observations
+        and other sites are regarded as latent variables over which a
+        posterior is to be inferred.
+    :param list target_labels: A subset of the sample sites over which the posterior
+        entropy is to be measured.
+    :param int N: Number of outer expectation samples.
+    :param int M: Number of inner expectation samples for `p(y|d)`.
+    :param int M_prime: Number of samples for `p(y | theta, d)` if required.
+    :return: EIG estimate
+    :rtype: `torch.Tensor`
+    """
+
+    if isinstance(observation_labels, str):#list of strings instead of strings
+        observation_labels = [observation_labels]
+    if isinstance(target_labels, str):
+        target_labels = [target_labels]
+        
+    expanded_design = lexpand(design, N, 1)#N copies of the model
+    shape = list(design.shape[:-1])
+    expanded_yspace = {k: rexpand(y, *shape) for k, y in yspace.items()}
```

Comment:
TODO: we might be able to read this out of the model, rather requiring the user to specify. We should also Eli